src/SubPool.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.26;
+
+import {Auth} from "./Auth.sol";
+import {ISubPoolFactory} from "./interfaces/ISubPoolFactory.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+import {ERC20} from "solady/tokens/ERC20.sol";
+
+contract SubPool is Auth {
+    error SubPool__OnlyFactory();
+    error SubPool__ExitDelayNotElapsedYet();
+    error SubPool__CollateralTokenAlreadyInitialized();
+    error SubPool__InsufficientETH();
+    error SubPool__CannotBillAfterExitDelay();
+    error SubPool__InvalidWithdraw();
+
+    /// @dev cannot be immutable because that allows for multiple pools per solver address.
+    address public collateralToken;
+    address public immutable COW;
+    ISubPoolFactory public immutable factory;
+
+    uint256 collateralDue;
+    uint256 cowDue;
+    uint256 ethDue;
+
+    constructor(address _owner, address _cow) {
+        factory = ISubPoolFactory(msg.sender);
+        COW = _cow;
+
+        // rely the owner
+        _addOwner(_owner);
+    }
+
+    function initializeCollateralToken(address token) external {
+        if (collateralToken != address(0)) revert SubPool__CollateralTokenAlreadyInitialized();
+        collateralToken = token;
+    }
+
+    modifier onlyFactory() {
+        if (msg.sender != address(factory)) {
+            revert SubPool__OnlyFactory();
+        }
+        _;
+    }
+
+    /// @notice Signal the intent to announce exit the solver pool.
+    function announceExit() external auth {
+        factory.announceExit();
+    }
+
+    /// @notice Execute the exit after the exit delay has elapsed.
+    function exit() external auth {
```

Comment:
This feels a bit too complex. (I particularly find confusing the indirection of how the factory and the pool exit conditions play together.)
What about not having an exit function at all?
Then the parameter `hasExited` can be replaced with the check `exitTimestamp != 0 && block.timestamp >= exitTimestamp`.
Transferring the collateral out can be handled by `withdrawTokens`.
The fast track lets the owner set an arbitrary exit timestamp.
The only drawback is that there would be no `Exit` event: at some point it will just be possible to take the collateral. I'd argue this isn't an issue because the critical event is the exit announcement, and a solver shouldn't be allowed to settle as soon as its exit is announced. Also, this is already an issue because technically you could never call `exit` and get back all your collateral through `withdrawTokens`.

Reply:
removed `hasExited` from state